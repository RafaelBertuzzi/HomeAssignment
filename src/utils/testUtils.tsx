import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
