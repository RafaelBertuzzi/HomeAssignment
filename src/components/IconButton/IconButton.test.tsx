import { cleanup, fireEvent, screen } from '@testing-library/react-native';
import { rendererWithProviders, renderWithProviders } from '@utils/testUtils';

import { IconButton } from '.';

const onPress = jest.fn();

describe('Component: IconButton', () => {
  afterEach(cleanup);

  it('should render icon', () => {
    const json = rendererWithProviders(
      <IconButton icon="search" onPress={onPress} />,
    );
    expect(json).toMatchSnapshot();
  });

  it('should press button', () => {
    renderWithProviders(<IconButton icon="search" onPress={onPress} />);

    fireEvent.press(screen.getByRole('button'));
    expect(onPress).toHaveBeenCalled();
  });
});
