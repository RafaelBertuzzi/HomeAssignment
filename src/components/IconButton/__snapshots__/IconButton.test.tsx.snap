// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IconButton should render icon 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 10,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={{}}
        style={{}}
      >
        <Image
          accessibilityLabel="search"
          alt="search"
          dataSet={{}}
          onError={[Function]}
          source={
            {
              "testUri": "../../../src/assets/icons/search.png",
            }
          }
          style={
            {
              "height": 24,
              "maxWidth": "100%",
              "width": 24,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
